src/services/pallets/PalletsStorageItemService.ts

```diff
@@ -0,0 +1,118 @@
+import {
+	BlockHash,
+	ModuleMetadataV11,
+	StorageEntryMetadataV11,
+} from '@polkadot/types/interfaces';
+import { BadRequest, InternalServerError } from 'http-errors';
+import { IPalletStorageItem } from 'src/types/responses/PalletStorageItem';
+
+import { AbstractService } from '../AbstractService';
+
+export class PalletsStorageItemService extends AbstractService {
+	async fetchStorageItem({
+		hash,
+		palletId,
+		storageItemId,
+		key1,
+		key2,
+		metadata,
+	}: {
+		hash: BlockHash;
+		palletId: string;
+		storageItemId: string;
+		key1: string | undefined;
+		key2: string | undefined;
+		metadata: boolean;
+	}): Promise<IPalletStorageItem> {
+		if (!this.api.query[palletId]) {
+			throw new BadRequest(
+				`Invalid palletId: "${palletId}". Currently only pallet names are expected as a palletId.`
```

Comment:
The error message sounds like there are potentially other ways of providing the `palletId`, what would those other ways be?
Also isn't it possible that the `palletId` is technically valid but there is no such pallet?

Reply:
> The error message sounds like there are potentially other ways of providing the palletId, what would those other ways be?

We are planning on supporting look ups by index for `pallets/id/events`, `pallets/id/errors`, and `pallets/id/calls`. I was thinking at some point in the future we might use the same for storage items. But thinking about it I am actually  not sure that makes sense / is a paradigm used in substrate.

> Also isn't it possible that the palletId is technically valid but there is no such pallet?
Good point. I'll try and think of some better phrasing

Reply:
Updated the error messages. Let me know what you think.